pan3d/explorers/analytics.py

```diff
@@ -0,0 +1,310 @@
+
+
+class Pan3dAnalyticsView(Pan3DView):
+    def __init__(self, render_window, **kwargs):
+        super().__init__(render_window=render_window, **kwargs)
+        with self.toolbar:
+            v3.VDivider(classes="my-1")
+            with v3.VTooltip(text="Display charting/plotting drawer"):
+                with html.Template(v_slot_activator="{ props }"):
+                    v3.VBtn(
+                        v_bind="props",
+                        flat=True,
+                        density="compact",
+                        icon="mdi-chart-line",
+                        click="plot_drawer = !plot_drawer",
+                    )
+
+
+class AnalyticsExplorer(Explorer):
+    def __init__(self, xarray=None, source=None, server=None, local_rendering=None):
+        """Create an instance of the AnalyticsExplorer class."""
+        super().__init__(xarray, source, server, local_rendering)
+
+        self.xarray = xarray
+        self.ui = None
+        self._setup_vtk()
+        self._build_ui()
+
+    # -------------------------------------------------------------------------
+    # VTK Setup
+    # -------------------------------------------------------------------------
+
+    def _setup_vtk(self):
+        self.renderer = vtkRenderer(background=(0.8, 0.8, 0.8))
+        self.interactor = vtkRenderWindowInteractor()
+        self.render_window = vtkRenderWindow(off_screen_rendering=1)
+
+        self.render_window.AddRenderer(self.renderer)
+        self.interactor.SetRenderWindow(self.render_window)
+        self.interactor.GetInteractorStyle().SetCurrentStyleToTrackballCamera()
+
+        self.lut = vtkLookupTable()
+        self.source = vtkXArrayRectilinearSource(input=self.xarray)
+
+        # Need explicit geometry extraction when used with WASM
+        self.geometry = vtkDataSetSurfaceFilter(
+            input_connection=self.source.output_port
+        )
+        self.mapper = vtkPolyDataMapper(
+            input_connection=self.geometry.output_port,
+            lookup_table=self.lut,
+        )
+        self.actor = vtkActor(mapper=self.mapper, visibility=0)
+
+        self.interactor.Initialize()
+
+        axes_actor = vtkAxesActor()
+        self.widget = vtkOrientationMarkerWidget()
+        self.widget.SetOrientationMarker(axes_actor)
+        self.widget.SetInteractor(self.interactor)
+        self.widget.SetViewport(0.85, 0, 1, 0.15)
+        self.widget.EnabledOn()
+        self.widget.InteractiveOff()
+
+    # -------------------------------------------------------------------------
+    # Trame API
+    # -------------------------------------------------------------------------
+
+    def start(self, **kwargs):
+        """Initialize the UI and start the server for XArray Viewer."""
+        self.ui.server.start(**kwargs)
+
+    @property
+    async def ready(self):
+        """Start and wait for the XArray Viewer corroutine to be ready."""
+        await self.ui.ready
+
+    @property
+    def state(self):
```

Comment:
Since you are an explorer, should those method be removed as they are part of our parent class?